### PR TITLE
Made tracing optional for preview environments

### LINKED ISSUE #334: Update gradle in gitpod/workspace-full
**Describe the bug**
Currently running `gradle init` in a fresh `gitpod/workspace-full` based workspace yields:
```
FAILURE: Build failed with an exception.

* What went wrong:
java.lang.ExceptionInInitializerError (no error message)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
```

**To Reproduce**
 1. Open a fresh repository
 1. run `gradle init --type java-library`

**Expected behavior**
Gradle should initialize the project.

It seems to be a [gradle incompatibility](https://stackoverflow.com/questions/42851099/java-lang-exceptionininitializererror-with-gradle) with java 8 (we have `jdk 1.8.0_201`). Updating to `gradle-4.10.3` solves the issues; sadly this is not available from the ubuntu repos.

### PR DESCRIPTION
**Pull request by wulfthimm**
## Description

This makes tracing optional for preview-environments and binds its deployment to the observability stack.
- werft-job without observability: https://werft.gitpod-dev.com/job/gitpod-build-wth-tracing-optional.5/raw
- werft-job with observability: https://werft.gitpod-dev.com/job/gitpod-build-wth-tracing-optional.7/raw

## Related Issue(s)

Fixes https://github.com/gitpod-io/ops/issues/334

## How to test

Start a prev-environment without and `with-observability`.

## Release Notes
```release-notes
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
@wulfthimm: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **princerachit** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @princerachit` in a comment when ready.

Associated issue: *#334*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: chart/templates/_helpers.tpl
Comment (mads-hartmann) [inline chart/templates/_helpers.tpl:261]: I think this might be a breaking change. I believe that all users of the Helm chart that previous has tracing enabled by setting `tracing.endpoint` would now have to update their values files to also have `tracing.enabled`

If so I would rather no go down this route and suggested in https://github.com/gitpod-io/ops/issues/334 and just use additional values files for preview environments.

**Review by mads-hartmann — CHANGES_REQUESTED** [review]
I believe this is a breaking change to the Helm chart. Please see comment ☺️

File: chart/templates/_helpers.tpl
Comment (wulfthimm) [reply chart/templates/_helpers.tpl:261]: I change it, but it is common to use `.enabled` to switch optional deployments.

**Comment by ArthurSens** [comment]
Wulf, our changelog automation relies on us setting the release notes to the PR description. Could you add it, please?

just add this to your PRs (Edit this comment and copy the content below):

```release-notes
NONE
```

**Comment by ArthurSens** [comment]
Also, I think you meant 

`Fixes https://github.com/gitpod-io/ops/issues/334` instead of `Fixes #334` 🙂

File: chart/templates/_helpers.tpl
Comment (wulfthimm) [reply chart/templates/_helpers.tpl]: Done

**Review by mads-hartmann — APPROVED** [review]
The code change looks good to me. I added the `hold` label so that Prow won't merge. Please verify that with your latest code change the error logs from e.g. ws-manger-bridge are not longer there as per the "How to reproduce" section of the [issue](https://github.com/gitpod-io/ops/issues/334)

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 0074564f274ce068c54c31c1735ef92c45dd507c</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7020#pullrequestreview-821503209" title="Approved">mads-hartmann</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **princerachit** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @princerachit` in a comment when ready.

Associated issue: *#334*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Review by ArthurSens — APPROVED** [review]
LGTM for me as well, now let's hunt for an OWNER's approval 😅

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7020#pullrequestreview-822592512" title="Approved">ArthurSens</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7020#pullrequestreview-821503209" title="Approved">mads-hartmann</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **princerachit** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @princerachit` in a comment when ready.

Associated issue: *#334*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by mads-hartmann** [comment]
/assign @princerachit

Per quest by roboquat

**Comment by princerachit** [comment]
Hi I could not fully understand this part of the description:
`...binds its deployment to the observability stack...`.
I can see that we are using the flag to control configuration of tracing for helm installation but don't understand how does it make tracing optional. Do you mean that monitoring satellite will continue to deploy tracing components like otel but due to this flag the actual metric will never travel to honeycomb?

**Comment by wulfthimm** [comment]
> 

If you add "with-observabiliy" you get tracing, if you do not then there will be no tracing at all.

**Comment by princerachit** [comment]
> If you add "with-observabiliy" you get tracing, if you do not then there will be no tracing at all.

Ok, so in abstraction I get it. But monitoring satellite will continue deploying the tracing components, right?

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7020#pullrequestreview-822592512" title="Approved">ArthurSens</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7020#pullrequestreview-821503209" title="Approved">mads-hartmann</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7020#issuecomment-986554277" title="Approved">princerachit</a>*

Associated issue: *#334*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [princerachit]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by mads-hartmann** [comment]
@princerachit You only get monitoring-satellite if you use `with-observability`. This PR fixes the Gitpod configuration so that tracing is only configured if we have somewhere to send them ☺️ The issue gitpod-io/ops#334 has a bit more context.

**Pull request merged**
